### ISSUE TITLE
Implement theme system.

### ISSUE THREAD

**Issue by nonzzz** [nonzzz/squarified#4]
### Describe the feature request

# Background

In `v0.1.x` we release a stable version but now i want to provide more powerful functions, like `cache`,`theme` and etc. It's a long task so i can't do them at same time. so this feature request i hope can do `font size` automatic adjustment and other render theme. ( Add new graph) 

The package size module must be tree-shakable.

**Comment by nonzzz** [comment]
## TL;DR
About this feature request. `v0.2.0` was published. In this version we do a lot of optimization for `tree-map component`, so about `theme` and etc i think it can be later for next few versions. Currently i want to make sure this component can work well for most browser ( I don't decide to support mobile or others, pc is enough.)

### Event System

The design of the event system is currently opinionated. It's not possible for now if someone wants to disable some events. (Because you might not disable all of event by your self.)

```ts

// Disable the drag 

treemap.off('mousemove')

``` 

So it need for more option to control the behavior.


### Others

I add some comment in `ARCHITECTURE.md`

**Comment by nonzzz** [comment]
#5 #6
